cardano-node/src/Cardano/Node/Handlers/Shutdown.hs

```diff
@@ -92,28 +129,40 @@ withShutdownHandling ShutdownConfig{scIPC = Just fd} tr action = do
 -- | Spawn a thread that would cause node to shutdown upon ChainDB reaching the
 -- configuration-defined slot.
 maybeSpawnOnSlotSyncedShutdownHandler
-  :: ShutdownConfig
+  :: HasHeader (Header blk)
+  => ShutdownConfig
   -> Tracer IO ShutdownTrace
   -> ResourceRegistry IO
   -> ChainDB.ChainDB IO blk
   -> IO ()
 maybeSpawnOnSlotSyncedShutdownHandler sc tr registry chaindb =
-  case scOnSlotSynced sc of
-    Just (MaxSlotNo maxSlot) -> do
-      traceWith tr (ShutdownArmedAtSlot maxSlot)
-      spawnSlotLimitTerminator maxSlot
-    _ -> pure ()
+  case scOnSyncLimit sc of
+    Nothing -> pure ()
+    Just lim -> do
+      traceWith tr (ShutdownArmedAt lim)
+      spawnLimitTerminator lim
  where
-  spawnSlotLimitTerminator :: SlotNo -> IO ()
-  spawnSlotLimitTerminator maxSlot =
+  spawnLimitTerminator :: SlotOrBlock Identity -> IO ()
+  spawnLimitTerminator limit =
     void $ forkLinkedWatcher registry "slotLimitTerminator" Watcher {
-        wFingerprint = id
+        wFingerprint = identity
```

Comment:
Why the change to `identity`?

Reply:
@Jimbo4350, sorry, `Cardano.Prelude` import confused me, undone.

Reply:
Why did you have to make this change? You shouldn't have to.

Reply:
I needed `Generic`, and so instead of adding `GHC.Generics`, I switched the prelude, which allowed dumping a bunch of imports: https://github.com/input-output-hk/cardano-node/pull/3932/commits/2c020c824970a9a8b31928dfc894f282630fe0e7#diff-9a71b006f6ddf2ab12e67e4d7e2951abf353bc8841f8a2638a1336a76b2ae43bL23-L32